src/scene/particle-container/shared/ParticleBuffer.ts

```
@@ -0,0 +1,227 @@
+    private _size = 0;
+    private readonly _dynamicUpload: ParticleUpdateFunction;
+    private readonly _staticUpload: ParticleUpdateFunction;
+    private readonly _generateParticleUpdateCache: Record<string, {
+        dynamicUpdate: ParticleUpdateFunction;
+        staticUpdate: ParticleUpdateFunction;
+    }> = {};
+
+    constructor(options: ParticleBufferOptions)
+    {
+        // size in sprites!
+        const size = this._size = options.size ?? 1000;
```

Biome

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)